docker-compose.microservices.example.yml

```diff
@@ -0,0 +1,117 @@
+      # L2 Node
+      - L2_NODE_WEB3_URL=http://geth_l2:8545                                            # The URL of the L2 node
+      # L1 Submitters
+      - L1_SEQUENCER_PRIVATE_KEY=<TODO: Add this>                                                   # (Required) The private key to use to submit Sequencer Transaction Batches
+      - L1_STATE_ROOT_PRIVATE_KEY=<TODO: Add this>                                                  # (Required) The private key to use to submit State Root Batches (will cause nonce issues if the same as the above key)
+      # Shared Contracts
+      - CANONICAL_TRANSACTION_CHAIN_CONTRACT_ADDRESS=<TODO: Add this>                               # (Required) The address of the CanonicalTransactionChain contract
+      - STATE_COMMITMENT_CHAIN_CONTRACT_ADDRESS=<TODO: Add this>                                    # (Required) The address of the StateCommitmentChain contract
+      # L1 Chain Data Persister (needs Postgres & L1 Node vars above)
+      - L1_TO_L2_TRANSACTION_QUEUE_CONTRACT_ADDRESS=<TODO: Add this>                                # (Required) The address of the L1ToL2TransactionQueue contract
+      - SAFETY_TRANSACTION_QUEUE_CONTRACT_ADDRESS=<TODO: Add this>                                  # (Required) The address of the SafetyTransactionQueue contract
+      - L1_CHAIN_DATA_PERSISTER_DB_PATH=/mnt/l1-node                                                # (Required) The filepath where to locate (or create) the L1 Chain Data Persister LevelDB database
+      - L1_EARLIEST_BLOCK=7001000                                                                   # (Required) The earliest block to sync on L1 to start persisting data
+      # L2 Chain Data Persister (needs Postgres & L2 Node vars above)
+      - L2_CHAIN_DATA_PERSISTER_DB_PATH=/mnt/l2-node                                                # (Required) The filepath where to locate (or create) the L2 Chain Data Persister LevelDB database
+      # Geth Submission Queuer (needs Postgres vars above)
+      - IS_SEQUENCER_STACK=1                                                                        # (Required) Set if this is queueing Geth submissions for a sequencer (and not _just_ a verifier)
+      - GETH_SUBMISSION_QUEUER_PERIOD_MILLIS                                                        # The period in millis at which the GethSubmissionQueuer should attempt to queue an L2 Geth submission (defaults to 10,000)
+      # Queued Geth Submitter (needs Postgres & L2 Node vars above)
+      - SUBMIT_TO_L2_PRIVATE_KEY=<TODO: Add this>                                                   # (Required) The PK to use to sign batches to send to geth (deprecated)
+      - QUEUED_GETH_SUBMITTER_PERIOD_MILLIS                                                         # The period in millis at which the QueuedGethSubmitter should attempt to send L2 Geth submissions (defaults to 10,000)
```

Comment:
difference between these two?

Reply:
Yep, they're two different services that poll the DB, so each one of them can, but doesn't have to, configure the period at which they poll. The first polls for L1 batches to queue as geth submissions, and the second reads from the queue and submits.

Reply:
Got it! makes sense :)